app/ante/eth.go

```diff
@@ -571,3 +542,22 @@ func (vbd EthValidateBasicDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simu
 
 	return next(ctx, tx, simulate)
 }
+
+// EthSetupContextDecorator is adapted from SetUpContextDecorator from cosmos-sdk, it ignores gas consumption
+// by setting the gas meter to infinite
+type EthSetupContextDecorator struct{}
+
+func NewEthSetUpContextDecorator() EthSetupContextDecorator {
+	return EthSetupContextDecorator{}
+}
+
+func (esc EthSetupContextDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
+	// all transactions must implement GasTx
+	_, ok := tx.(authante.GasTx)
```

Comment:
can you add a unit test for a transaction that fails this check?

Reply:
sure, added